aws_lambda_powertools/utilities/parser/models/kinesis_firehose_sqs.py

```diff
@@ -0,0 +1,29 @@
+import json
+from typing import List, Optional
+
+from pydantic import BaseModel, PositiveInt, validator
+
+from aws_lambda_powertools.shared.functions import base64_decode
+from aws_lambda_powertools.utilities.parser.models import KinesisFirehoseRecordMetadata
+
+from .sqs import SqsRecordModel
+
+
+class KinesisFirehoseSqsRecord(BaseModel):
+    data: SqsRecordModel
+    recordId: str
+    approximateArrivalTimestamp: PositiveInt
+    kinesisRecordMetadata: Optional[KinesisFirehoseRecordMetadata]
+
+    @validator("data", pre=True, allow_reuse=True)
+    def data_base64_decode(cls, value):
+        # Firehose payload is encoded twice
```

Comment:
😱

Reply:
Ruben, I will validate this one more time before merging. But scare me too.